app/controllers/PSRController.scala

```diff
@@ -105,6 +107,10 @@ abstract class PSRController extends FrontendBaseController with I18nSupport {
         }
     }
 
+  def memberPaymentsChanged(implicit request: DataRequest[_]): Boolean = request.pureUserAnswers.exists(
+    !_.sameAs(request.userAnswers, membersPayments, Omitted.membersPayments: _*)
```

Comment:
Maybe I'm missing something here, but this doesn't consider the index of the member, so if you have multiple members and any of the members has changed and e.g. you navigate to CYA page for a member that hasn't been changed and submit the CYA page then memberStatus get flipped to "Changed" (and by that version for that member increased too).

Reply:
Thats true but we omit MemberStatus from the comparison, it reality it's the old comparison logic that Cem wrote but I had to omit fields like MemberStatus for that very reason

In hindsight we probably should have built a case class with all member payments fields pulled from UA (like we do when we compare members for memberPsrVersion)

Reply:
I probably couldn't describe the problem properly, so let's have a chat about it.

Reply:
Fixed

Reply:
`PSRController.memberPaymentsChanged` still being called from multiple places in member payments and some member-index data set based on it. E.g.  `MemberPensionPaymentsCYAController.onSubmit`, `EmployerContributionsCYAController.onSubmit`, and 5 other places. Would we have a similar bug if e.g. member payment is CYA page is reached on the UI and submitted without a change?

Reply:
Fixed now, thanks for that 👍